client/client.go

```diff
@@ -144,6 +147,24 @@ func New(cfg Config) (Client, error) {
 	return c, nil
 }
 
+// KeepSync keeps synchronizing with etcd cluster about its member list,
+// so client could update its endpoints and catch up the change of the cluster.
+func KeepSync(ctx context.Context, c Client) error {
```

Comment:
Can we rename this to autoSync, make client a receiver and add interval as a parameter?
Basically I propose for `(c Client) AutoSync(ctx context.Context, interval time.Duration)`

Comments can be much simpler.
// AutoSync periodically calls Sync() every given interval.